prereise/gather/demanddata/transportation_electrification/data_process.py

```diff
@@ -0,0 +1,132 @@
+                else:
+                    sorted_data.iloc[i, sorted_data.columns.get_loc('Dwell time (hour decimal)')] = 24 - \
+                        sorted_data.iloc[i-1, sorted_data.columns.get_loc('End time (hour decimal)')] + sorted_data.iloc[i-total_trips-1, sorted_data.columns.get_loc('Start time (hour decimal)')]
+        
+        else:
+            sorted_data.iloc[i-total_trips:i, sorted_data.columns.get_loc('sample vehicle number')] = sample_veh_num
+            sorted_data.iloc[i-total_trips:i, sorted_data.columns.get_loc('total vehicle trips')] = total_trips
+            sorted_data.iloc[i-total_trips:i, sorted_data.columns.get_loc('total vehicle miles travelled')] = total_miles
+            
+            sample_veh_num += 1
+
+            if total_trips == 1:
+                sorted_data.iloc[i-1, sorted_data.columns.get_loc('Dwell time (hour decimal)')] = 24 - \
+                    (sorted_data.iloc[i-1, sorted_data.columns.get_loc('Start time (hour decimal)')] - sorted_data.iloc[i-1, sorted_data.columns.get_loc('End time (hour decimal)')])
+            else:
+                sorted_data.iloc[i-1, sorted_data.columns.get_loc('Dwell time (hour decimal)')] = 24 - \
+                    sorted_data.iloc[i-1, sorted_data.columns.get_loc('End time (hour decimal)')] + sorted_data.iloc[i-total_trips-1, sorted_data.columns.get_loc('Start time (hour decimal)')]
+            
+            total_trips = 1
+            total_miles = sorted_data.iloc[i, sorted_data.columns.get_loc('Vehicle miles travelled')]
+           
+            curr_vehicle_id = sorted_data.iloc[i, sorted_data.columns.get_loc('Vehicle ID')]
+            curr_household = sorted_data.iloc[i, sorted_data.columns.get_loc('Household')]
+
+            # this makes sure the last entry gets included; 
+            # otherwise, it would break out of loop w/o adding 
+            if i == (l-1):
+                sorted_data.iloc[i, sorted_data.columns.get_loc('total vehicle trips')] = total_trips
+                sorted_data.iloc[i, sorted_data.columns.get_loc('total vehicle miles travelled')] = total_miles
+                sorted_data.iloc[i, sorted_data.columns.get_loc('sample vehicle number')] = sample_veh_num
+
+                if total_trips == 1:
+                    sorted_data.iloc[i, sorted_data.columns.get_loc('Dwell time (hour decimal)')] = 24 - \
+                        (sorted_data.iloc[i-1, sorted_data.columns.get_loc('Start time (hour decimal)')] - sorted_data.iloc[i-1, sorted_data.columns.get_loc('End time (hour decimal)')])
+                else:
+                    sorted_data.iloc[i, sorted_data.columns.get_loc('Dwell time (hour decimal)')] = 24 - \
+                        sorted_data.iloc[i-1, sorted_data.columns.get_loc('End time (hour decimal)')] + sorted_data.iloc[i-total_trips-1, sorted_data.columns.get_loc('Start time (hour decimal)')]
+
+        i += 1
+
+    sorted_data.to_excel("output.xlsx")
```

Comment:
We probably want to make this step optional, since I don't think we use Excel files downstream.

Reply:
This was actually more for me when testing things. I can go ahead and remove it